src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -215,6 +226,15 @@ public async Task InitializeUIAffinitizedServicesAsync(IAsyncServiceProvider asy
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_threadingContext.DisposalToken);
 
             openFileTracker.ProcessQueuedWorkOnUIThread();
+
+            // Switch to a background thread to avoid loading option providers on UI thread.
```

Comment:
Comment seems not to match the code, unless the point is fetching the telemetry stuff is also implicitly triggering option provider loading? If so, the comment might be worth expanding on.

Reply:
Telemetry is using options. I'll add more comment here.